diff -aur minbif-1.0.5.pristine/CMakeLists.txt minbif-1.0.5.new/CMakeLists.txt
--- minbif-1.0.5.pristine/CMakeLists.txt	2011-12-04 12:24:51.000000000 -0200
+++ minbif-1.0.5.new/CMakeLists.txt	2012-07-07 22:20:43.952609999 -0300
@@ -67,13 +67,13 @@
 			MESSAGE(FATAL_ERROR "You need to enable CACA to enable VIDEO")
 		ENDIF(NOT ENABLE_CACA)
 		PKG_CHECK_MODULES(GSTREAMER gstreamer-0.10>=0.10.10 gstreamer-interfaces-0.10>=0.10)
-		PKG_CHECK_MODULES(FARSIGHT farsight2-0.10>=0.0.9)
+		PKG_CHECK_MODULES(FARSTREAM farstream-0.1>=0.0.9)
 		PKG_CHECK_MODULES(PURPLE_26 purple>=2.6.1)
-		IF(GSTREAMER_FOUND AND PURPLE_26_FOUND AND FARSIGHT_FOUND)
+		IF(GSTREAMER_FOUND AND PURPLE_26_FOUND AND FARSTREAM_FOUND)
 			SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DHAVE_VIDEO")
-		ELSE(GSTREAMER_FOUND AND PURPLE_26_FOUND AND FARSIGHT_FOUND)
-			MESSAGE(FATAL_ERROR "Dependencies for video were not met. Install gstreamer, farsight and libpurple>=2.6 first. Or to disable video, run 'make ENABLE_VIDEO=0'")
-		ENDIF(GSTREAMER_FOUND AND PURPLE_26_FOUND AND FARSIGHT_FOUND)
+		ELSE(GSTREAMER_FOUND AND PURPLE_26_FOUND AND FARSTREAM_FOUND)
+			MESSAGE(FATAL_ERROR "Dependencies for video were not met. Install gstreamer, farstream and libpurple>=2.6 first. Or to disable video, run 'make ENABLE_VIDEO=0'")
+		ENDIF(GSTREAMER_FOUND AND PURPLE_26_FOUND AND FARSTREAM_FOUND)
 	ENDIF(ENABLE_VIDEO)
 
 	OPTION(ENABLE_PAM "Enable PAM support" ON)
@@ -124,8 +124,8 @@
 	PKG_CHECK_MODULES(LIBXML REQUIRED libxml-2.0>=2.5)
 ENDIF(ENABLE_PLUGIN)
 
-INCLUDE_DIRECTORIES(${PURPLE_INCLUDE_DIRS} ${GTHREAD_INCLUDE_DIRS} ${CACA_INCLUDE_DIRS} ${IMLIB_INCLUDE_DIRS} ${GSTREAMER_INCLUDE_DIRS} ${FARSIGHT_INCLUDE_DIRS} ${LIBXML_INCLUDE_DIRS} ${PAM_INCLUDE_DIRS} ${GNUTLS_INCLUDE_DIRS} "src/")
-LINK_DIRECTORIES(${PURPLE_LIBRARY_DIRS} ${GTHREAD_LIBRARY_DIRS} ${CACA_LIBRARY_DIRS} ${IMLIB_LIBRARY_DIRS} ${GSTREAMER_LIBRARY_DIRS} ${FARSIGHT_LIBRARY_DIRS} ${LIBXML_LIBRARY_DIRS} ${GNUTLS_LIBRARY_DIRS})
+INCLUDE_DIRECTORIES(${PURPLE_INCLUDE_DIRS} ${GTHREAD_INCLUDE_DIRS} ${CACA_INCLUDE_DIRS} ${IMLIB_INCLUDE_DIRS} ${GSTREAMER_INCLUDE_DIRS} ${FARSTREAM_INCLUDE_DIRS} ${LIBXML_INCLUDE_DIRS} ${PAM_INCLUDE_DIRS} ${GNUTLS_INCLUDE_DIRS} "src/")
+LINK_DIRECTORIES(${PURPLE_LIBRARY_DIRS} ${GTHREAD_LIBRARY_DIRS} ${CACA_LIBRARY_DIRS} ${IMLIB_LIBRARY_DIRS} ${GSTREAMER_LIBRARY_DIRS} ${FARSTREAM_LIBRARY_DIRS} ${LIBXML_LIBRARY_DIRS} ${GNUTLS_LIBRARY_DIRS})
 
 SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D_REENTRANT -D_FILE_OFFSET_BITS=64 -Wall -Wextra -Wno-unused-parameter")
 SET(CMAKE_CXX_FLAGS ${CMAKE_C_FLAGS})
diff -aur minbif-1.0.5.pristine/src/CMakeLists.txt minbif-1.0.5.new/src/CMakeLists.txt
--- minbif-1.0.5.pristine/src/CMakeLists.txt	2011-12-04 12:24:51.000000000 -0200
+++ minbif-1.0.5.new/src/CMakeLists.txt	2012-07-07 22:20:43.952609999 -0300
@@ -56,7 +56,7 @@
 		irc/conversation_channel.cpp
 	      )
 
-TARGET_LINK_LIBRARIES(${BIN_NAME} "-lpthread -lstdc++" ${PURPLE_LIBRARIES} ${GTHREAD_LIBRARIES} ${CACA_LIBRARIES} ${IMLIB_LIBRARIES} ${GSTREAMER_LIBRARIES} ${FARSIGHT_LIBRARIES} ${PAM_LIBRARIES} ${GNUTLS_LIBRARIES})
+TARGET_LINK_LIBRARIES(${BIN_NAME} "-lpthread -lstdc++" ${PURPLE_LIBRARIES} ${GTHREAD_LIBRARIES} ${CACA_LIBRARIES} ${IMLIB_LIBRARIES} ${GSTREAMER_LIBRARIES} ${FARSTREAM_LIBRARIES} ${PAM_LIBRARIES} ${GNUTLS_LIBRARIES})
 
 INSTALL(TARGETS ${BIN_NAME}
         DESTINATION bin)
diff -aur minbif-1.0.5.pristine/src/im/account.cpp minbif-1.0.5.new/src/im/account.cpp
--- minbif-1.0.5.pristine/src/im/account.cpp	2011-12-04 12:24:51.000000000 -0200
+++ minbif-1.0.5.new/src/im/account.cpp	2012-07-07 22:20:43.959276650 -0300
@@ -18,6 +18,7 @@
 
 #include <cassert>
 #include <cstring>
+#include <unistd.h>
 #ifdef HAVE_IMLIB
 	#include <Imlib2.h>
 #endif /* HAVE_IMLIB */
diff -aur minbif-1.0.5.pristine/src/im/request.cpp minbif-1.0.5.new/src/im/request.cpp
--- minbif-1.0.5.pristine/src/im/request.cpp	2011-12-04 12:24:51.000000000 -0200
+++ minbif-1.0.5.new/src/im/request.cpp	2012-07-07 22:20:43.959276650 -0300
@@ -18,6 +18,7 @@
 
 #include <errno.h>
 #include <string.h>
+#include <unistd.h>
 
 #include "request.h"
 #include "buddy.h"
diff -aur minbif-1.0.5.pristine/src/irc/dcc.cpp minbif-1.0.5.new/src/irc/dcc.cpp
--- minbif-1.0.5.pristine/src/irc/dcc.cpp	2011-12-04 12:24:51.000000000 -0200
+++ minbif-1.0.5.new/src/irc/dcc.cpp	2012-07-07 22:20:43.959276650 -0300
@@ -23,6 +23,7 @@
 #include <string.h>
 #include <sys/types.h>
 #include <netinet/in.h>
+#include <unistd.h>
 
 #include "dcc.h"
 #include "nick.h"
diff -aur minbif-1.0.5.pristine/src/server_poll/daemon_fork.cpp minbif-1.0.5.new/src/server_poll/daemon_fork.cpp
--- minbif-1.0.5.pristine/src/server_poll/daemon_fork.cpp	2011-12-04 12:24:51.000000000 -0200
+++ minbif-1.0.5.new/src/server_poll/daemon_fork.cpp	2012-07-07 22:20:43.959276650 -0300
@@ -20,7 +20,7 @@
 #include <cassert>
 #include <cstring>
 #include <cerrno>
-#include <glib/gmain.h>
+#include <glib.h>
 #include <sys/socket.h>
 #include <sys/stat.h>
 #include <arpa/inet.h>
diff -aur minbif-1.0.5.pristine/src/server_poll/inetd.cpp minbif-1.0.5.new/src/server_poll/inetd.cpp
--- minbif-1.0.5.pristine/src/server_poll/inetd.cpp	2011-12-04 12:24:51.000000000 -0200
+++ minbif-1.0.5.new/src/server_poll/inetd.cpp	2012-07-07 22:20:43.959276650 -0300
@@ -17,7 +17,8 @@
  */
 
 #include <cassert>
-#include <glib/gmain.h>
+#include <glib.h>
+#include <unistd.h>
 
 #include "inetd.h"
 #include "irc/irc.h"
diff -aur minbif-1.0.5.pristine/src/sockwrap/sockwrap.cpp minbif-1.0.5.new/src/sockwrap/sockwrap.cpp
--- minbif-1.0.5.pristine/src/sockwrap/sockwrap.cpp	2011-12-04 12:24:51.000000000 -0200
+++ minbif-1.0.5.new/src/sockwrap/sockwrap.cpp	2012-07-07 22:20:43.962609976 -0300
@@ -16,6 +16,8 @@
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
  */
 
+#include <unistd.h>
+
 #include "sockwrap.h"
 #include "sockwrap_plain.h"
 #ifdef HAVE_TLS
